src/uu/tail/src/tail.rs

```diff
@@ -54,73 +58,80 @@ impl Default for Settings {
 
 #[allow(clippy::cognitive_complexity)]
 pub fn uumain(args: impl uucore::Args) -> i32 {
-    let args = args.collect_str();
+    let program = args.next().unwrap_or_else(|| OsString::from(NAME));
+    let args = std::iter::once(program.clone()).chain(args);
 
     let mut settings: Settings = Default::default();
 
-    // handle obsolete -number syntax
-    let options = match obsolete(&args[1..]) {
-        (args, Some(n)) => {
-            settings.mode = FilterMode::Lines(n, b'\n');
-            args
-        }
-        (args, None) => args,
-    };
-
-    let args = options;
-
-    let mut opts = getopts::Options::new();
-
-    opts.optopt("c", "bytes", "Number of bytes to print", "k");
-    opts.optopt("n", "lines", "Number of lines to print", "k");
-    opts.optflag("f", "follow", "Print the file as it grows");
-    opts.optopt(
-        "s",
-        "sleep-interval",
-        "Number or seconds to sleep between polling the file when running with -f",
-        "n",
-    );
-    opts.optopt(
-        "",
-        "pid",
-        "with -f, terminate after process ID, PID dies",
-        "PID",
-    );
-    opts.optflag("z", "zero-terminated", "Line delimiter is NUL, not newline");
-    opts.optflag("h", "help", "help");
-    opts.optflag("V", "version", "version");
-    opts.optflag("v", "verbose", "always output headers giving file names");
-    opts.optflag("q", "quiet", "never output headers giving file names");
-    opts.optflag("", "silent", "synonym of --quiet");
-
-    let given_options = match opts.parse(&args) {
-        Ok(m) => m,
-        Err(_) => {
-            println!("{}", opts.usage(""));
-            return 1;
-        }
-    };
-
-    if given_options.opt_present("h") {
-        println!("{}", opts.usage(""));
-        return 0;
-    }
-    if given_options.opt_present("V") {
-        version();
-        return 0;
-    }
-
-    settings.follow = given_options.opt_present("f");
+    let mut app = App::new(executable!())
+        .version(crate_version!())
+        .about("output the last part of files")
+        .arg(
+            Arg::with_name("bytes")
```

Comment:
please follow the patterns used elsewhere for consistency :+1: 
option commands are declared using variables:
https://github.com/uutils/coreutils/blob/master/src/uu/chown/src/chown.rs#L30-L41

Reply:
I will extract the variables for the option commands.

On a related note, and I was going to ask initially, but is it allowed to combine `with_name(OPT_NO_PRESERVE_ROOT).long(OPT_NO_PRESERVE_ROOT)` into a single cusotm macro! call? Otherwise the code is really repetitive.

Reply:
not after aware of :)

Reply:
Aware of what? Do I change it?

Reply:
Done the needful.